azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -905,19 +911,15 @@ public void resume(String user) {
   }
 
   public void kill(String user) {
-    synchronized (mainSyncObj) {
-      logger.info("Flow killed by " + user);
-      flow.setStatus(Status.KILLED);
-      kill();
-      updateFlow();
-    }
-    interrupt();
+    logger.info("Flow killed by " + user);
+    kill();
   }
 
-  private void kill() {
+  public void kill() {
     synchronized (mainSyncObj) {
+      if(flowKilled) return;
       logger.info("Kill has been called on flow " + execId);
-
+      flow.setStatus(Status.KILLED);
```

Comment:
`setStatus` just sets the status right? Should we set it once all the process has been killed? This seems to be more wishful where we assume that we will kill the flow in the upcoming steps.